### ISSUE TITLE
Identity Resolution

### ISSUE THREAD

**Issue by ahankinson** [music-encoding/music-encoding#112]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on September 18, 2012 13:30:31_

Currently avref expects a link to a specific audio file. However, there are times where you will not be able to resolve an audio file. For example, if you create a time-aligned corpus of audio to chord symbols and want to share it broadly, you can share the annotations but not share the audio (due to copyright restrictions).

To get around this, we should expand the <timeline> element to support other audio/video identity resolution schemes. This could be schemes that use fingerprinting and unique ids, like MusicBrainz, or standardized references like the International Standard Recording Code ( http://en.wikipedia.org/wiki/International_Standard_Recording_Code )

It may look something like this:

<timeline
  identity=" http://musicbrainz.org/recording/ebf79ba5-085e-48d2-9eb8-2d992fbf0f6d ">
  ...
</timeline>

@-identity is something I just made up -- we could probably re-use an existing attribute on this element.

_Original issue: http://code.google.com/p/music-encoding/issues/detail?id=112_

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on September 18, 2012 11:51:37_

@-avref was intended to point to an <avFile> element; that is, a URI that begins with '#'.  But, since it's defined as data.URI, it could refer to anything accessible by a URI, including the musicbrainz URL.

<timeline avref=" http://musicbrainz.org/recording/ebf79ba5-085e-48d2-9eb8-2d992fbf0f6d ">

if that's what you wanted to do.  However, it seems logical to me to associate the other schemes you suggest with <avFile>, not <timeline>. For example,

<timeline origin="#w1" avref="#a1">
  <when xml:id="w1"/>
  <when xml:id="w2"/>
  <when xml:id="w3"/>
</timeline>
<performance>
  <recording>
    <avFile xml:id="a1" target=" http://musicbrainz.org/recording/ebf79ba5-085e-48d2-9eb8-2d992fbf0f6d"/ >
  </recording>
</performance>

In this case, <avFile> is not a specific audio/video file, but _a reference_ to an a/v file.  This is not unlike <graphic target=" http://blah.blah.blah/imageCGI?image01"/ > where the actual image is "hidden behind" a reference.

The advantage of the extra level of indirection is that <avFile>, <recording>, or <performance> can be associated with metadata provided in the header, while <timeline> cannot.

Does this accomplish what you're looking for?

**Comment by ahankinson** [comment]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on September 18, 2012 12:57:43_

That's the general idea, although I worry about adding a song identity as @-avref. The song identity almost never resolves to an audio file; it just serves as a way of saying "the song I have in my hand is the same recording as the song that was used for the timepoint reference."

So when I'm generating the MEI I would use the MusicBrainz fingerprinter to find the MusicBrainz URL for the audio that I'm using, and would store that in the MEI file. When you receive my file you would have to then de-reference it by running the same fingerprinter on your audio. If the MusicBrainz URL you get from the fingerprinter is the same as the one that I stored you can be reasonably sure that the track you have is the same as the one I used to generate it.

Using @-avfile, or even <avFile /> would also mean I couldn't store a reference to an audio file as well as the identity, if I wanted.

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on October 04, 2012 15:38:44_

Your references to "song", "audio", and "recording" are confusing to me.  First, you say a "song" is not an audio file, but then you say "the _song I have in my hand_ is the _same recording_ as the song that was used for the timepoint reference."  Does this sentence not say that song = recording?  What is it you have in your hand?  ;-)

Is the MusicBrainz URL is a identifier for a particular recording/song?  Or is it an identifier for a work/song?  That is, does it identify Elvis Presley's recording of "Blue Suede Shoes" or all recordings of "Blue Suede Shoes"?

In either case, I believe the MusicBrainz identifier itself is metadata that belongs somewhere other than on <timeline>; that is, in/on <source>.  The question is, where's the best place to put a pointer to that metadata?

**Comment by ahankinson** [comment]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on October 05, 2012 01:49:47_

You have an MEI file containing chord symbols with timepoints corresponding to a recording you have of Blue Suede Shoes. I want this data for my own research purposes, but because of copyright restrictions you cannot share the audio file you have with me.

Now, I have a thousand different versions of Blue Suede Shoes on my local computer -- live recordings, re-recordings, cover songs, re-orchestrations, etc. There is a very good possibility that you don't NEED to share your audio with me -- I probably have a version that will work just fine. 

But how do you tell me which version you used? You cannot give me an audio file. You cannot simply tell me "Elvis Presley's 'Blue Suede Shoes'" because he recorded fifty different versions of it. 

So you give me the MusicBrainz identifier for the recording. To get this you run the MusicBrainz fingerprinter (AcoustID) on the audio file you have. This will give you a fingerprint. You submit this fingerprint to MusicBrainz, which then tells you that it corresponds to this record: http://musicbrainz.org/recording/cae0281b-210b-4cd4-94db-c09be4b3277c You then store this identifier in your MEI file and hand it off to me.

I receive the MEI file with this identifier. I run the same fingerprinter on my own collection of audio files and receive a unique fingerprint for each of the audio files. I submit each fingerprint to MusicBrainz, and, lo and behold, one of them returns the same Identifier! I can then use that audio file as the source for the @-avref in the timeline, but you have not shared the audio with me.

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on October 10, 2012 13:11:18_

Ok, the MusicBrainz identifier identifies a particular recording.  So, one possible place to store the MBid is here

<recording xml:id="cae0281b-210b-4cd4-94db-c09be4b3277c">

This sets up a one-to-one correspondence between the recording element and the MBid.

But this is using a generic attribute for a specific purpose, which may make the encoding unintelligible to other users/applications. It's better to put the MBid elsewhere, perhaps on a source element or in the content of a source/identifier element. The recording element can point to the source using its @-decls attribute.

This is somewhat "round about", but is entirely in keeping with the function of recording and source.  The recording element is analogous to surface in the facsimile module and shouldn't be used to capture metadata about a particular recorded (for lack of a better term, bibliographic) entity.  That's the job of source.

I'm aware that this is confusing. Perhaps renaming recording to something else would help clear up some of the confusion, but we (Raffaele, Johannes, and I) haven't been able to come up with a better term.  Suggestions will be greatly appreciated.

BTW, I haven't seen any of your comments in my in box, only after I visit here.  Was this on purpose or was the "send email" box not checked in error?

**Comment by ahankinson** [comment]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on October 12, 2012 02:24:20_

I've unchecked the e-mail box, but I thought it would still go to the participants of this discussion. I'll leave it checked.

Yeah, I really disagree with using xml:id to store this information, not least because you may want to refer to the same MBID twice in the same file.

I like the <source /> element solution. Could <recording /> have @-source, which points to a source ID?

Then on <source /> we could maybe have @-target?

**Comment by ahankinson** [comment]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on October 17, 2012 07:52:52_

**Labels:** -Milestone-Release2013 Milestone-ReleaseNext

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on February 12, 2013 13:44:33_

Seems like I dropped the ball on this one a long time ago.  <recording> has @decls by which any element in the header, not just <source> can be associated with the <recording> element.  <source> provides @-data for recording pointers (note the plural) back in the other direction.

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on June 09, 2014 14:35:09_

While I'm reluctant to add a lot of metadata to <recording>, I can see that adding att.authorized (@authority & @authURI) could provide a quick path to a standard identifier such as the MusicBrainz ID. However, other metadata about the recording such as title, date/place of issue, etc., would still need to be in a <source> element. The <recording> and <source> elements can be linked together via their @-decls attributes, the <recording> element pointing to the xml:id of the <source> and/or the <source> element pointing to the xml:id of the <recording>.

**Comment by ahankinson** [comment]
_From [andrew.hankinson](https://code.google.com/u/andrew.hankinson/) on September 12, 2014 04:41:19_

Closing since this is outdated. I'm sure this can be done with the new FRBR module if one were really dedicated.

**Status:** WontFix
